src/Command/GenerateEnvironmentCommand.php

```diff
@@ -155,23 +172,32 @@ protected function execute(InputInterface $input, OutputInterface $output) {
         'environment_url' => $this->validateUrl($input->getOption('environment-url')),
         'backup_base' => $input->getOption('backup-db') ? 1 : 0,
         'reimport' => $input->getOption('reimport') ? 1 : 0,
-        'fetch_dump' => $input->getOption('fetch-dump') ? 1 : 0,
-        'dump_file_name' => $input->getOption('dump-file-name'),
+        'dump_always_update' => $input->getOption('dump-always-update') ? 1 : 0,
       ];
 
-      if ($generateParams['fetch_dump']) {
-        $generateParams += [
-          'ssh_config_name' => $input->getOption('ssh-config-name'),
-          'ssh_dump_path' => $input->getOption('ssh-dump-path'),
-        ];
+      if ($input->getOption('dump-retrieval-tool') == 'scp') {
+        if (!empty($input->getOption('ssh-config-name'))) {
+          $scp_connection = $input->getOption('ssh-config-name');
+        }
+        else {
+          $scp_connection = $input->getOption('scp-connection-info');
+        }
+        $recap_fetch_command = 'scp ' . $scp_connection . ':' . $input->getOption('fetch-source-path') . ' ' . $this->generator->getCombawaRoot() . '/' . $input->getOption('fetch-dest-path');
+      }
+      else {
+        $recap_fetch_command = 'cp ' . $input->getOption('fetch-source-path') . ' ' . $this->generator->getCombawaRoot() . '/' . $input->getOption('fetch-dest-path');
       }
+      $generateParams += [
+        'dump_fetch_command' => $recap_fetch_command,
+        'dump_file_name' => $input->getOption('fetch-dest-path'),
+      ];
```

Comment:
As I told you. I don't like to have this command generated in a variable then executed by the script as-is.
The previous version where combawa was selecting which tool to use and how was probably flawed too but I'd rather improve the old version than replace it by this one. That being said, I won't fight too much for this so the final decision is up to you.

Reply:
Follow up dans #59 + #58.